### PR TITLE
Improve bounding area

### PR DESCRIPTION
**Pull request by blackleg**
Get north latitude and south latitude of the bounding area.
Get west longitude and east longitude of the bounding area.
Add statics metods to build new objects. 
Add more setters and getters.

### CODE REVIEW

**Comment by gr211** [comment]
Hi, sorry just saw your commits. Thanks for that. Will take a look very shortly.

File: src/main/java/com/grum/geocalc/BoundingArea.java [hunk 1]
```diff
@@ -131,4 +139,72 @@ public int hashCode() {
...
         return hash;
     }
+    
+    /**
+     * Get North Latitude
+     * @return double latitude
+     */
+    public double getNorthLatitude() {
+        return northEast.getLatitude();
+    }
```
Comment (gr211) [inline src/main/java/com/grum/geocalc/BoundingArea.java]: this and al. are only used in the same class (and associated tests) and do not bring much value. If anything, that clutters  the code quite a bit.

File: src/main/java/com/grum/geocalc/BoundingArea.java [hunk 2]
```diff
@@ -131,4 +139,72 @@ public int hashCode() {
...
+
+    /**
+     * Get South Latitude
+     * @return double latitude
+     */
+    public double getSouthLatitude() {
+        return southWest.getLatitude();
```
Comment (gr211) [inline src/main/java/com/grum/geocalc/BoundingArea.java]: same as above

File: src/main/java/com/grum/geocalc/BoundingArea.java [hunk 3]
```diff
@@ -131,4 +139,72 @@ public int hashCode() {
...
+    }
+    
+    /**
+     * Get the lower latitude coordinate value
+     * @return
+     */
+    public double getLowerLatitudeValue() {
+        if (getSouthLatitude() > getNorthLatitude()) {
```
Comment (gr211) [inline src/main/java/com/grum/geocalc/BoundingArea.java]: I would tend to favor Math.min() and Math.max() methods

File: src/main/java/com/grum/geocalc/BoundingArea.java [hunk 4]
```diff
@@ -131,4 +139,72 @@ public int hashCode() {
...
+            return getNorthLatitude();
+        } else {
+            return getSouthLatitude();
+        }
+    }
+    
+    public double getHigherLatitudeValue() {
+        if (getSouthLatitude() > getNorthLatitude()) {
```
Comment (gr211) [inline src/main/java/com/grum/geocalc/BoundingArea.java]: same as above

File: src/main/java/com/grum/geocalc/BoundingArea.java [hunk 5]
```diff
@@ -131,4 +139,72 @@ public int hashCode() {
...
+            return getNorthLatitude();
+        }
+    }
+    
+    /**
+     * Get West Longitude
+     * @return double
+     */
+    public double getWestLongitude() {
+        return southWest.getLongitude();
```
Comment (gr211) [inline src/main/java/com/grum/geocalc/BoundingArea.java]: same as above

File: src/main/java/com/grum/geocalc/BoundingArea.java [hunk 6]
```diff
@@ -131,4 +139,72 @@ public int hashCode() {
...
+    }
+    
+    /**
+     * Get East Longitude
+     * @return double
+     */
+    public double getEastLongitude() {
+        return northEast.getLongitude();
```
Comment (gr211) [inline src/main/java/com/grum/geocalc/BoundingArea.java]: same as above

File: src/main/java/com/grum/geocalc/BoundingArea.java [hunk 7]
```diff
@@ -131,4 +139,72 @@ public int hashCode() {
...
+    }
+    
+    public double getLowerLongitudeValue() {
+        if (getWestLongitude() > getEastLongitude()) {
```
Comment (gr211) [inline src/main/java/com/grum/geocalc/BoundingArea.java]: same as above

File: src/main/java/com/grum/geocalc/BoundingArea.java [hunk 8]
```diff
@@ -131,4 +139,72 @@ public int hashCode() {
...
+            return getEastLongitude();
+        } else {
+            return getWestLongitude();
+        }
+    }
+    
+    public double getHigherLongitudeValue() {
+        if (getWestLongitude() > getEastLongitude()) {
```
Comment (gr211) [inline src/main/java/com/grum/geocalc/BoundingArea.java]: same as above

File: src/main/java/com/grum/geocalc/Point.java [hunk 9]
```diff
@@ -47,6 +48,27 @@ public Point(Coordinate latitude, Coordinate longitude) {
...
+    
+    
+    /**
+     * Build new point with latitude and longitude
+     * @param latitude
+     * @param longitude
+     * @return 
+     */
+    public static Point build(Coordinate latitude, Coordinate longitude) {
+        return new Point(latitude, longitude);
```
Comment (gr211) [inline src/main/java/com/grum/geocalc/Point.java]: falls a bit short of a useful builder pattern

File: src/main/java/com/grum/geocalc/Point.java [hunk 10]
```diff
@@ -47,6 +48,27 @@ public Point(Coordinate latitude, Coordinate longitude) {
...
+    }
+    
+    /**
+     * Build new point with latitude and longitude
+     * @param latitude
+     * @param longitude
+     * @return Point
+     */
+    public static Point build(double latitude, double longitude) {
+        return build(DegreeCoordinate.build(latitude), DegreeCoordinate.build(longitude));
```
Comment (gr211) [inline src/main/java/com/grum/geocalc/Point.java]: I would object here as well. It looses the type-safety of what a double value means (ie is it degrees, radians ?). The javadoc could indicate it but it currently does not

File: src/main/java/com/grum/geocalc/DegreeCoordinate.java [hunk 11]
```diff
@@ -43,4 +43,13 @@
...
         this.decimalDegrees = decimalDegrees;
     }
+    
+    /**
+     * Build new <code>DegreeCoordinate</code>
+     * @param decimalDegrees 
+     * @return <code>DegreeCoordinate</code>
+     */
+    public static DegreeCoordinate build(double decimalDegrees) {
+        return new DegreeCoordinate(decimalDegrees);
```
Comment (gr211) [inline src/main/java/com/grum/geocalc/DegreeCoordinate.java]: same as above

File: src/main/java/com/grum/geocalc/Point.java [hunk 12]
```diff
@@ -56,6 +78,20 @@ public Point(Coordinate latitude, Coordinate longitude) {
     public double getLatitude() {
         return latitude;
     }
+    
+    public Coordinate getLatitudeCoordinate() {
+        return DegreeCoordinate.build(latitude);
+    }
+
+    public void setLatitude(double latitude) {
+        this.latitude = latitude;
```
Comment (gr211) [inline src/main/java/com/grum/geocalc/Point.java]: also the whole point of not having setter was to make those Point objects immutable.

File: src/main/java/com/grum/geocalc/Point.java [hunk 13]
```diff
@@ -65,6 +101,20 @@ public double getLatitude() {
...
+    public Coordinate getLongitudeCoordinate() {
+        return DegreeCoordinate.build(longitude);
+    }
+
+    public void setLongitude(double longitude) {
+        this.longitude = longitude;
+    }
+    
+    public void setLongitude(Coordinate longitude) {
+        if (Objects.nonNull(longitude)) {
```
Comment (gr211) [inline src/main/java/com/grum/geocalc/Point.java]: same as above about immutability

**Comment by gr211** [comment]
Thanks @blackleg. I am not too sure these changes would be for the better. They break immutability, add not-so-useful builder pattern like constructs with removed type-safety, and it appears they is a misunderstanding of what higher/lower longitudes might mean.

**Comment by blackleg** [comment]
@gr211 I try to resolve some of the questions.

**Pull request closed without merging**